Remove trailing whitespace from docs files

awly opened the pull request:
Some editors will do this automatically on save. This causes a lot of
diffs when editing the docs in such an editor.
Clean them up once now and we'll try to keep it tidy going forward.

webvictim reviewed (approved):
LGTM. 

Is there a command or one-liner you would recommend using to do this automatically?

I recently installed an extension in my editor (VS Code) which highlights trailing whitespace too: https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces

benarent reviewed (approved):
👍 but please don't merge yet, I'm going to do some other plumping to the 4.3 branch today and I know in the past it's got me cherry picking. I'll give you final thumbs up today.

benarent commented:
^ Ok, we are good to go. You might want to rebase and this should resolve the conflicts.

Commit:
Remove trailing whitespace from docs files

Some editors will do this automatically on save. This causes a lot of
diffs when editing the docs in such an editor.
Clean them up once now and we'll try to keep it tidy going forward.

awly commented:
> Is there a command or one-liner you would recommend using to do this automatically?

This does the trick for a directory:
```sh
find docs/4.3 -type f | xargs sed -i -E 's/\s+$//g'
```

Should we do it on `make docs` automatically for all versions?

webvictim commented:
> Should we do it on `make docs` automatically for all versions?

That sounds pretty good to me, what @benarent thinks is more relevant though as he does more regular changes. It'd be nice if we could do it as the same time as we run `milv` to validate the links.

In the new Drone CI world we can probably have a conditional job that runs on `docs/**` PRs and fails if there's extra whitespace.

benarent commented:
Thanks gents, I'm always open to improving my whitespace game.. I'm the likely culprit for adding them. 

Let's add this into a make file, that'll run Milv and cleanup whitespace at the same time.

Commit:
Add make rules for docs whitespace and milv

docs-test-whitespace: checks for trailing whitespace in all .md files
  under docs/.
docs-fix-whitespace: removes trailing whitespace in all .md files under
  docs/.
docs-test-links: runs milv in all docs/ subdirectories that have
  milv.config.yaml.
docs-test: runs whitespace and links tests, used during `make docs`

awly commented:
ok, added make rules for testing/fixing whitespace and milv.
`make docs` will test for whitespace and run milv.
`make docs-fix-whitespace` fixes trailing whitespace.
we can add `make docs-test` into our Jenkins config too.

PTAL @benarent @webvictim

benarent commented on Makefile:
```diff
@@ -190,6 +190,43 @@ docs:
+docs-test-whitespace:
+	if find docs/ -type f -name '*.md' | xargs grep -E '\s+$$'; then \
+		echo "trailing whitespace found in docs/ (see above)"; \
+		echo "run 'make docs-fix-whitespace' to fix it"; \
```
👍 very handy input.

benarent commented on Makefile:
```diff
@@ -190,6 +190,43 @@ docs:
+docs-test-links: DOCS_FOLDERS := $(shell find -name milv.config.yaml | xargs dirname)
+docs-test-links:
+	go get -v github.com/magicmatatjahu/milv
+	for docs_dir in $(DOCS_FOLDERS); do \
```
When we introduced Milv we didn't run it on older versions, since now moved out older version and only have current the mass milv fix should be must easier. Once merged, I'll fix the rest of the output.
